### ISSUE TITLE
Improvement: Support user-assigned country codes.

### ISSUE THREAD

**Issue by derekmahar** [TakahikoKawasaki/nv-i18n#25]
Please support [user-assigned country codes](http://en.wikipedia.org/wiki/ISO_3166-1_numeric#User-assigned_code_elements).  For example, field Country Identifier in a [Global Intermediary Identification Number](http://www.irs.gov/PUP/businesses/corporations/giin_composition.pdf) supports country number 999 for "Other".

**Comment by TakahikoKawasaki** [comment]
Do you mean that "Other" should be added to CountryCode? Or, that CountryCode methods such as getByCode(int) should treat 900-999 differently? Or that some new methods for user-assigned country codes should be added to CountryCode?

Could you clarify what the support means?

**Comment by derekmahar** [comment]
"Other" is just an example. I think the implication is that CountryCode
would no longer be a strictly immutable enumeration, but would also contain
a map of numeric or alpha codes to some immutable instance of a user
defined class. Perhaps this would have to be a separate "union" class, say
CountryOrUserAssignedCode that would accept user assignments and would have
it's own getByCode(int) method that would return a standard or
user-assigned code.

**Comment by TakahikoKawasaki** [comment]
Okay, I understood your intention. Are you in a hurry for this extension?

**Comment by derekmahar** [comment]
No, not yet, but I was going to implement a workaround myself in a week or
two.

**Comment by mihxil** [comment]
How about something like https://github.com/mihxil/i18n-regions

**Comment by mihxil** [comment]
I released a 'i18n-regions' which supports user assigned countries, while using nv-i18n for official countries.
